File: 231-power-of-two/power-of-two.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isPowerOfTwo = function(n) {
    if(n==1) return true;
    if(n<=0 || n%2!==0) return false;

     let x = Math.log2(n);
     return Number.isInteger(x);
     
    // if(2**x == n) return true;
    if(Math.pow(2,x)==n) return true;
    
};

File: 231-power-of-two/power-of-two.java
class Solution {
    public boolean isPowerOfTwo(int n) {
        

        // if(n<=0) return false;

        // double power = Math.log(n)/Math.log(2.0);
        // int roundedPower = (int) Math.round(power);  

        // if(Math.pow(2,roundedPower)==n) return true;
        // else return false;

        if(n==1) return true;
        if(n<=0 || n%2!=0) return false;
         
        return isPowerOfTwo(n/2);

    }
}